File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        // 소문자 처리
		String lowerString = s.toLowerCase();
		StringBuilder sb = new StringBuilder();
		for(char c : lowerString.toCharArray()){
			// 영숫자 확인
			if(Character.isLetterOrDigit(c)) sb.append(c);
		}

		String a = sb.toString();
		String b = sb.reverse().toString();

		return a.equals(b);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var start = 0;
        var end = s.length - 1;

        while (start < end) {
            when {
                !Character.isLetterOrDigit(s[start]) -> start++
                !Character.isLetterOrDigit(s[end]) -> end--
                else -> {
                    if (s[start].lowercaseChar() != s[end].lowercaseChar()) {
                        println(s[start].lowercaseChar())
                        println(s[end].lowercaseChar())
                        return false
                    }
                    start++
                    end--
                }
            }
        }
        return true
    }
}